includes/polling/ospf.inc.php

```diff
@@ -138,6 +139,35 @@
         ->whereNotIn('id', $ospf_neighbours->pluck('id'))->delete();
 
     echo $ospf_neighbours->count();
+
+    echo ' TOS Metrics: ';
+
+    // Pull data from device
+    $ospf_tos_poll = snmpwalk_cache_oid($device, 'OSPF-MIB::ospfIfMetricEntry', [], 'OSPF-MIB');
+    d_echo($ospf_tos_poll);
+
+    $ospf_tos_metrics = collect();
+    foreach ($ospf_tos_poll as $ospf_tos_id => $ospf_tos) {
+        // get ospf_port_id
+        $ospf_tos['ospf_port_id'] = OspfPort::query()
+            ->where('ospfIfIpAddress', $ospf_tos['ospfIfMetricIpAddress'])
+            ->where('context_name', $device['context_name'])
+            ->value('ospf_port_id');
+        $tos = OspfTos::updateOrCreate([
+            'device_id' => $device['device_id'],
+            'ospf_port_id' => $ospf_tos['ospf_port_id'],
+            'context_name' => $device['context_name'],
+        ], $ospf_tos);
+
+        $ospf_tos_metrics->push($tos);
+    }
+
```

Comment:
Why aren't you graphing these?

Reply:
I tried copying the RRD code for OSPF TOS, but my rrdcached was giving an error for unknown arguments. Can you add additional datasets to an RRD file after it's created?

Reply:
No, you cannot modify the datasets in an RRD.  You would need a new rrd file just for this.

Reply:
Right. Route costing is set in the router config, so something like config versioning would let someone tracking cost over time. I don't think we need a separate RRD graph for it.

Reply:
sounds good